CairoMakie/src/overrides.jl

```diff
@@ -12,6 +12,12 @@ function draw_plot(scene::Scene, screen::Screen, poly::Poly)
     draw_poly(scene, screen, poly, to_value.(poly.input_args)...)
 end
 
+# Override `get_all_plots` to allow `poly` to remain a unit,
+# instead of auto-decomposing in lines and mesh.
+function get_all_plots(plot::Poly, plots = AbstractPlot[])
```

Comment:
it would be nice if we could just re-use `Makie.flatten_plots` .. Maybe we can add an attribute `flatten_plots(... ; is_atomic_plot=p-> isempty(p.plots)`, so that CairoMakie can pass a different definition of what it considers atomic?